calculator.go

```diff
@@ -1,13 +1,99 @@
 // Package calculator provides a library for simple calculations in Go.
 package calculator
 
-// Add takes two numbers and returns the result of adding them together.
-func Add(a, b float64) float64 {
-return a + b
+import (
+	"errors"
+	"fmt"
+	"math"
+)
+
+// Add takes two or more numbers and returns the result of adding them together.
+func Add(a, b float64, nums ...float64) float64 {
+	var sum float64 = a + b
+	for _, num := range nums {
+		sum += num
+	}
+	return sum
 }
 
-// Subtract takes two numbers and returns the result of subtracting the second
+// Subtract takes two or more numbers and returns the result of subtracting the second
 // from the first.
-func Subtract(a, b float64) float64 {
-	return b - a
+func Subtract(a, b float64, nums ...float64) float64 {
+	var total float64 = a - b
+	for _, num := range nums {
+		total -= num
+	}
+	return total
+}
+
+// Multiply takes two or more numbers and returns the result of multiplying them together
+func Multiply(a, b float64, nums ...float64) float64 {
+	var total float64 = a * b
+	for _, num := range nums {
+		total *= num
+	}
+	return total
+}
+
+// Divide takes two or more numbers and returns the result of dividing a by b
+func Divide(a, b float64, nums ...float64) (float64, error) {
+	if b == 0 {
+		return 0, errors.New("divide by zero")
+	}
+	// if the first number is zero then the result is zero.
+	if a == 0 {
+		return 0, nil
+	}
+
```

Comment:
The Nth commandment of BIT is "thou shalt not write blank lines within functions", for small values of N (I haven't worked out the exact priority order yet).